File: ruby/resistor-color/resistor_color.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/resistor-color
# Resistor Color exercise
module ResistorColor
  COLORS = %w[
    black
    brown
    red
    orange
    yellow
    green
    blue
    violet
    grey
    white
  ].freeze

  def self.color_code(color)
    COLORS.index(color)
  end
end

File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

resistor_band_t color_code(resistor_band_t color) { return color; }

const resistor_band_t *colors(void) {
    static resistor_band_t colors_list[NUMBER_OF_COLORS] = {BLACK};

    for (int i = 0; i < NUMBER_OF_COLORS; i++) {
        colors_list[i] = (resistor_band_t)i;
    }

    return colors_list;
}
